easytensor/src/Numeric/Matrix/Mat44d.hs

```diff
@@ -0,0 +1,196 @@
+{-# LANGUAGE MagicHash #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Numeric.Matrix.Mat44d () where
+
+import qualified Control.Monad.ST as ST
+import GHC.Exts
+import Numeric.DataFrame.Internal.Array.Family.DoubleX3
+import Numeric.DataFrame.Internal.Array.Family.DoubleX4
+import qualified Numeric.DataFrame.ST as ST
+import Numeric.DataFrame.SubSpace
+import Numeric.DataFrame.Type
+import Numeric.Matrix.Class
+import Numeric.Scalar
+import Numeric.Vector
+
+{-# INLINE mat44d #-}
+mat44d ::
+  Scd -> Scd -> Scd -> Scd ->
+  Scd -> Scd -> Scd -> Scd ->
+  Scd -> Scd -> Scd -> Scd ->
+  Scd -> Scd -> Scd -> Scd ->
+  Mat44d
+mat44d
+  _11 _12 _13 _14
+  _21 _22 _23 _24
+  _31 _32 _33 _34
+  _41 _42 _43 _44
+  = ST.runST $ do
```

Comment:
Looks like a very handy function! Makes the rest of the code very readable.

As a matter of optimization, we can use function [`PrimArray.gen#`](https://github.com/achirkin/easytensor/blob/0cac23e2f5de16670f6bbc35e59117f1a4c200f2/easytensor/src/Numeric/DataFrame/Internal/Array/Class.hs#L21) (which operates on packed `Double`),
or even manually write the primitive array to avoid repacking primitive `Double#`.

In theory, GHC should be able to do worker/wrapper transform if optimization enabled. If it is consistently performed on `ST.writeDataFrameOff` and if `mat44d` is always inlined, the module should end up with no repacking of primitive values at all and be as fast as we could possible do manually. Probably, we can check this by looking at the generated code, but it is not very pleasant thing to do :)